### ISSUE TITLE
Fork sphinx-js

### ISSUE THREAD

**Issue by hoodmane** [pyodide/pyodide#3549]
Sphinx-js is super helpful in generating our API docs but it isn't really maintained. For instance, it only works with typedoc v0.17 which is nearly three years out of date. We work around this by heavily patching it. I propose to make a new `pyodide/sphinx-js` fork and use this fork.

I asked the pyscript people because they also have typescript api docs and @FabioRosado indicated interest in working together on this.

**Comment by ryanking13** [comment]
Yeah, I think we need much more than what sphinx-js provides for now, so I'm +1 for forking it. Thanks for improving the docs by the way, I haven't been doing much jobs on improving docs, but I'll try to involve more after we fork sphinx-js.

Probably we should inform sphinx-js about why we are forking, what our goal is, what can be upstreamed, and what cannot be upstreamed, etc.

**Comment by hoodmane** [comment]
I think mostly the reason is that upstream isn't maintained. E.g., my PR improving support for newer versions of typedoc https://github.com/mozilla/sphinx-js/pull/219 got no response.
By contrast with sphinx-autodoc-typehints we needed a bunch of patches but they accepted all of them quickly.

**Comment by rth** [comment]
Sounds good. Do you anticipate publishing it to PyPI as well? In which case we should probably use a different name.

**Comment by hoodmane** [comment]
Yeah I was thinking publish it as pyodide-sphinx-js.

**Comment by raucao** [comment]
I just ran into the bug you fixed and it's pretty demotivating that nobody even replied until today. Any news on this potential fork being published? Couldn't find it on PyPI yet...

**Comment by hoodmane** [comment]
Hi @raucao no pypi release yet. My fork is here:
https://github.com/pyodide/sphinx-js/

I guess I should publish a release based on what I have since it's already a fair amount. I'm never really sure what to do with author metadata copyright etc when publishing a fork.

There's also more bug fixes / features that Pyodide patches in that I haven't moved into sphinx-js yet. I have a list of todos here:
https://github.com/pyodide/sphinx-js/issues/26

**Comment by jstasiak** [comment]
Hey all,

I came here from the `sphinx-js` repository and I'm wondering – is your fork expected to be usable/useful outside of `pyodide`? My impression is yes, but I'd like to confirm.

Thank you for your work.

**Comment by hoodmane** [comment]
I hope so.
